### PR TITLE
chore(deps): bump recharts from 2.4.3 to 2.15.4

### PR DESCRIPTION
**Pull request by chaoran-chen**
This updates Recharts to the last 2.x version (updating recharts to 3 shall be done later: #1111).

### CODE REVIEW

File: src/widgets/__tests__/__snapshots__/EstimatedCasesChart.test.tsx.snap
Comment (chaoran-chen) [inline src/widgets/__tests__/__snapshots__/EstimatedCasesChart.test.tsx.snap:1]: I've updated the tests on my device and they work consistently, so it seems that there might be device-dependent behavior..

Comment (chaoran-chen) [reply src/widgets/__tests__/__snapshots__/EstimatedCasesChart.test.tsx.snap:1]: The following tests are failing:

- VariantTimeDistributionLineChart.test.tsx
- EstimatedCasesChart.test.tsx

Comment (chaoran-chen) [reply src/widgets/__tests__/__snapshots__/EstimatedCasesChart.test.tsx.snap:1]: The differences are in the SVG paths:

image

File: src/widgets/EstimatedCasesChartInner.tsx [hunk 1]
```diff
@@ -139,7 +139,7 @@ export const EstimatedCasesChartInner = React.memo(
                     width={50}
                   />
                   <Tooltip
-                    active={false}
+                    active={true}
```
Comment (anna-parker) [inline src/widgets/EstimatedCasesChartInner.tsx:142]: I see you have switched to active true a lot in this PR - what is the behavior change?

File: src/widgets/Metrics.tsx [hunk 2]
```diff
@@ -97,33 +97,23 @@ export const ChartAndMetrics = ({
...
-
   return (
     <div id={`chart-and-metrics-${title}`} className='flex flex-col h-full w-full'>
       <h3 id={`chart-title-${title}`} className='my-0 pb-4 pr-10 pt-0 text-gray-500'>
         {title}
         <div>{buttons}</div>
       </h3>
-      <div className={childrenParentClass}>
-        <div className={childrenClass}>{children}</div>
+      <ChartAndMetricsWrapper>
```
Comment (anna-parker) [inline src/widgets/Metrics.tsx:108]: Do we have a test for this - or could you add a screenshot/recording that the behavior is the same?

File: src/widgets/__tests__/__snapshots__/HospitalizationDeathChart.test.tsx.snap [hunk 3]
```diff
@@ -27,21 +27,8 @@ exports[`<HospitalizationDeathChart> (died plot) dataset1 renders correctly 1`]
...
-                  "height": "100%",
-                  "maxHeight": undefined,
-                  "minHeight": undefined,
-                  "minWidth": undefined,
-                  "width": "100%",
-                }
-              }
-            />
-          </div>
+            className="sc-ewnqHT bRaosl"
```
Comment (anna-parker) [inline src/widgets/__tests__/__snapshots__/HospitalizationDeathChart.test.tsx.snap:30]: why have the `recharts-responsive-container`s been removed and why has the ID of the className changed?

File: src/widgets/EstimatedCasesChartInner.tsx (hunk 1 above)
Comment (chaoran-chen) [reply src/widgets/EstimatedCasesChartInner.tsx:142]: As far as I can tell, the user-observable behavior for our application remains the same. Previously, the tooltip side effects worked worked while `active` is set to false, now it doesn't anymore.

File: src/widgets/Metrics.tsx (hunk 2 above)
Comment (chaoran-chen) [reply src/widgets/Metrics.tsx:108]: We have snapshot tests that cover most of the charts and include the `ChartAndMetrics`. Everything looks fine as far as I can see (including the maps). I think the risks that something is entirely broken is very low. If something is out-of-place in an edge case, we can fix it when we observe it.

File: src/widgets/__tests__/__snapshots__/HospitalizationDeathChart.test.tsx.snap (hunk 3 above)
Comment (chaoran-chen) [reply src/widgets/__tests__/__snapshots__/HospitalizationDeathChart.test.tsx.snap:30]: We are now mocking (i.e., essentially excluding) recharts in `jest.mock('recharts');`. Previously, we included the outer container of `recharts-responsive-container` (but not the actual charts) but the old tests don't work anymore due to internal changes of recharts related to observing resizings and these tests were not very useful anyways.

Comment (chaoran-chen) [reply src/widgets/__tests__/__snapshots__/HospitalizationDeathChart.test.tsx.snap:30]: The IDs are from `styled-components` and I think they changed because I made some changes to the metrics.

**Comment by chaoran-chen** [comment]
I've used this now for a while as part of #1119. I'm mostly confident that this is alright and will therefore go ahead and merge it now.
